src/main/java/com/fakng/fakngagrgtr/parser/stripe/StripeParser.java

```diff
@@ -0,0 +1,156 @@
+        int pageCount = getPageCount(page);
+        List<Vacancy> vacancyList = new ArrayList<>();
+
+        for (int i = 0; i <= pageCount; i++) {
+            page = downloadPage(String.format(url, i));
+            vacancyList.addAll(fillVacancy(
+                    page.getByXPath(VACANCIES_URLS_XPATH), page.getByXPath(VACANCY_CITIES_XPATH)
+            ));
+        }
+
+        return vacancyList;
+    }
+
+    private int getPageCount(HtmlPage page) {
+        String url = ((DomElement) page.getByXPath(JOBS_PAGINATION_LIST_XPATH).get(0)).getAttribute("href");
+        return Integer.parseInt(url.replaceAll("\\D+", "")) / pageSize;
+    }
+
+    private List<Vacancy> fillVacancy(List<DomNode> urls, List<DomElement> cities) {
+        List<Vacancy> vacancies = new ArrayList<>();
+
+        for (int i = 0; i < urls.size(); i++) {
+            String jobId = urls.get(i).getAttributes().getNamedItem("href").getNodeValue();
+
+            // Skip repeating vacancies and parse locations for them
+            Set<LocationDTO> locationDTOs = new HashSet<>();
+            while (jobId.equals(urls.get(i).getAttributes().getNamedItem("href").getNodeValue())) {
+                locationDTOs.add(createLocationDto(cities.get(i)));
+                i++;
+            }
+
+            // Variable i stopped in incorrect position
+            i--;
+            vacancies.add(createVacancy(createVacancyDTO(jobId, locationDTOs)));
+        }
+
+        return vacancies;
+    }
+
+    @SneakyThrows
+    private VacancyDTO createVacancyDTO(String jobUrl, Set<LocationDTO> locationDTOS) {
+        VacancyDTO vacancyDTO = new VacancyDTO();
+
+        HtmlPage page = downloadPage(STRIPE_URL + jobUrl);
+        DomElement title = (DomElement) page.getByXPath(VACANCY_TITLE_XPATH).get(0);
+        DomElement jobType = ((DomElement) page.getByXPath(VACANCY_JOB_TYPE_XPATH).get(0)).getNextElementSibling();
+        DomElement team = ((DomElement) page.getByXPath(VACANCY_TEAM_XPATH).get(0)).getNextElementSibling();
+
+        vacancyDTO.setDescription(createDescription(page));
+        vacancyDTO.setUrl(STRIPE_URL + jobUrl);
+        vacancyDTO.setJobId(jobUrl.replaceAll("\\D+", ""));
+        vacancyDTO.setTeam(team.getTextContent());
+        vacancyDTO.setLocation(locationDTOS);
+        vacancyDTO.setJobType(jobType.getTextContent());
+        vacancyDTO.setTitle(title.getTextContent());
+
+        return vacancyDTO;
+    }
+
+    private Vacancy createVacancy(VacancyDTO vacancyDTO) {
+        Vacancy vacancy = new Vacancy();
+
+        vacancy.setTitle(vacancyDTO.getTitle());
+        vacancy.setDescription(createFullDescription(vacancyDTO));
+        vacancy.setUrl(vacancyDTO.getUrl());
+        vacancy.setJobId(vacancyDTO.getJobId());
+        vacancy.setCompany(company);
+        processLocations(vacancy, vacancyDTO.getLocation());
+
+        return vacancy;
+    }
+
+    private void processLocations(Vacancy vacancy, Set<LocationDTO> locations) {
```

Comment:
Please, get rid of Vacancy param here. Use setLocations method of the class for adding all locations that will be produced by this method.